src/rpc/server.h

```diff
@@ -42,6 +44,10 @@ class JSONRPCRequest
     void parse(const UniValue& valRequest);
 };
 
+inline void RPCHelpMan::Check(const JSONRPCRequest& request) const {
+    if (request.fHelp || !IsValidNumArgs(request.params.size())) throw std::runtime_error(ToString());
+}
```

Comment:
In commit 09e176a958  Make the RPCHelpMan aware of JSONRPCRequest and add Check() helper


Looks like you added the implementation to an unrelated header file (should be in rpc/util.cpp).

You can solve the circular dependency by moving JSONRPCRequest to rpc/protocol.h

Reply:
That sounds like a good solution, yeah. Thanks, doing that.